package com.pratik.job.job_portal_user_service.payload;

import com.pratik.job.domain.UserRole;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class SignupRequest {

    @NotBlank(message = "full name is mandatory")
    private String fullName;

    @Email(message = "Email should be valid")
    @NotBlank(message = "Email is mandatory")
    private String email;

    @NotBlank(message = "Password is madatory")
    private String password;

    private String phone;

    @NotNull(message = "Role is mandatory")
    private UserRole role;
}
